homeassistant/components/webdav/__init__.py

```diff
@@ -0,0 +1,71 @@
+"""The WebDAV integration."""
+
+from __future__ import annotations
+
+import logging
+
+from aiowebdav2.client import Client
+from aiowebdav2.exceptions import UnauthorizedError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME, CONF_VERIFY_SSL
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+from .const import CONF_BACKUP_PATH, DATA_BACKUP_AGENT_LISTENERS
+from .helpers import async_create_client, async_ensure_path_exists
+
+type WebDavConfigEntry = ConfigEntry[Client]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: WebDavConfigEntry) -> bool:
+    """Set up WebDAV from a config entry."""
+    client = async_create_client(
+        hass=hass,
+        url=entry.data[CONF_URL],
+        username=entry.data[CONF_USERNAME],
+        password=entry.data[CONF_PASSWORD],
+        verify_ssl=entry.data.get(CONF_VERIFY_SSL, True),
+    )
+
+    try:
+        result = await client.check()
+    except UnauthorizedError as err:
+        raise ConfigEntryAuthFailed("Invalid username or password") from err
```

Comment:
```suggestion
        raise ConfigEntryError("Invalid username or password") from err
```
until we have a reauth flow